File: commons/src/main/java/ru/starksoft/commons/numeric/NumericUtils.java
package ru.starksoft.commons.numeric;

import androidx.annotation.Nullable;

@SuppressWarnings("WeakerAccess")
public final class NumericUtils {

    private NumericUtils() {
        throw new UnsupportedOperationException();
    }

    public static int parseInt(@Nullable String input) {
        int result = 0;

        if (input == null || input.length() == 0) {
            return result;
        }

        try {
            result = Integer.parseInt(input.trim());
        } catch (NumberFormatException ignored) {
        }

        return result;
    }

    public static float parseFloat(@Nullable String input) {
        float result = 0;

        if (input == null || input.length() == 0) {
            return result;
        }

        try {
            result = Float.parseFloat(input.trim().replaceAll(",", "."));
        } catch (NumberFormatException ignored) {
        }

        return result;
    }


}

File: commons/src/main/java/ru/starksoft/commons/numeric/NumericUtils.kt
package ru.starksoft.commons.numeric

object NumericUtils {

    @JvmStatic
    fun parseInt(input: String?): Int {
        if (input.isNullOrBlank()) {
            return 0
        }
        return try {
            input.trim().toInt()
        } catch (ignored: NumberFormatException) {
            0
        }
    }

    @JvmStatic
    fun parseFloat(input: String?): Float {
        if (input.isNullOrBlank()) {
            return 0f
        }
        return try {
            input.trim().replace(",".toRegex(), ".").toFloat()
        } catch (ignored: NumberFormatException) {
            0f
        }
    }
}
